examples/36_form/src/components/App.tsx

```diff
@@ -16,7 +16,7 @@ const App = ({ name }: { name: string }) => {
           <h1>Hello {name}!!</h1>
           <h3>This is a server component.</h3>
           <Counter increment={increment} />
-          <Form message={getMessage()} greet={greet} />
+          <Form message={await getMessage()} greet={greet} />
```

Comment:
Wait, this change is very different. I think it's intentional to not have `await`.

Reply:
I was seeing errors in the e2e tests without this because it was attempting to render the unawaited promise which threw an error.

could you checkout the first commit of this PR and see if there's a different fix I should do?

Reply:
it didn't cause issue with 36_form I was just matching the behavior after seeing issues with it in the demo, but I am happy to revert just 36_form if you think it is still correct after checking

Reply:
Let me check.

Reply:
`
Error: Objects are not valid as a React child (found: object with keys {page:/, layout:/, root, route:/, ROUTE, IS_STATIC}). If you meant to render a collection of children, use an array instead.
`

It feels like a bug somewhere. But, I can't spot it yet.

Reply:
it looks like the object that we use here: https://github.com/dai-shi/waku/blob/main/packages/waku/src/router/client.ts#L477-L481

So maybe we are crossing paths with the promise for the Router and the getMessage promise? idk how such a thing would happen though

Reply:
Also if it's helpful. It's reproducible locally on the first load of the / route after starting the dev server. Then you need to restart the server to see the issue after that.

Reply:
If we disable SSR, the error won't happen. So, I assume it's SSR related.

Reply:
Spent the whole day and just found out how to reproduce the error without waku/router.

```diff
diff --git a/examples/36_form/src/components/App.tsx b/examples/36_form/src/components/App.tsx
index afb38d4f..4091d6ba 100644
--- a/examples/36_form/src/components/App.tsx
+++ b/examples/36_form/src/components/App.tsx
@@ -15,7 +15,9 @@ const App = ({ name }: { name: string }) => {
         >
           <h1>Hello {name}!!</h1>
           <h3>This is a server component.</h3>
+          {/*
           <Counter increment={increment} />
+          */}
           <Form message={getMessage()} greet={greet} />
           <ServerForm />
         </div>
```

That reproduces the error with 36_form.